rust/src/storage/file/storage_with_eviction_cache.rs

```diff
@@ -0,0 +1,577 @@
+
+        let result = storage.get(id).unwrap();
+        assert_eq!(**result.read().unwrap(), **node.read().unwrap());
+        assert!(storage.cache.get(&id).is_none());
+    }
+
+    #[test]
+    fn get_returns_not_found_error_if_id_is_present_as_delete_op() {
+        let storage = StorageWithEvictionCache {
+            cache: Arc::new(DashMap::new()),
+            storage: Arc::new(MockFileStorageManager::<MockFileBackend>::new()),
+            eviction_workers: EvictionWorkers {
+                workers: Vec::new(),
+                shutdown: Arc::new(AtomicBool::new(false)),
+            },
+        };
+
+        let id = NodeId::from_idx_and_node_type(0, NodeType::Inner);
+        storage.cache.insert(id, Op::Delete);
+
+        let result = storage.get(id);
+        assert!(matches!(result, Err(Error::NotFound)));
+    }
+
+    #[test]
+    fn get_returns_node_from_storage_if_not_in_cache() {
+        let id = NodeId::from_idx_and_node_type(0, NodeType::Inner);
+        let node = Node::Inner(Box::default());
+
+        let mut mock_storage = MockFileStorageManager::<MockFileBackend>::new();
+        mock_storage
+            .expect_get()
+            .withf(move |arg| *arg == id)
+            .returning({
+                let node = node.clone();
+                move |_| Ok(node.clone())
+            });
+
+        let storage_with_eviction_cache = StorageWithEvictionCache {
+            cache: Arc::new(DashMap::new()),
+            storage: Arc::new(mock_storage),
+            eviction_workers: EvictionWorkers {
+                workers: Vec::new(),
+                shutdown: Arc::new(AtomicBool::new(false)),
+            },
+        };
+
+        let result = storage_with_eviction_cache.get(id).unwrap();
+        assert_eq!(**result.read().unwrap(), node);
+    }
+
+    #[test]
+    fn reserve_retrieves_id_from_underlying_storage_layer_and_removes_from_cache() {
+        let id = NodeId::from_idx_and_node_type(0, NodeType::Inner);
+        let node = Node::Inner(Box::default());
+
+        let mut mock_storage = MockFileStorageManager::<MockFileBackend>::new();
+        mock_storage
+            .expect_reserve()
+            .withf({
+                let node = node.clone();
+                move |arg| arg == &node
+            })
+            .returning(move |_| id);
+
+        let storage_with_eviction_cache = StorageWithEvictionCache {
+            cache: Arc::new(DashMap::new()),
+            storage: Arc::new(mock_storage),
+            eviction_workers: EvictionWorkers {
+                workers: Vec::new(),
+                shutdown: Arc::new(AtomicBool::new(false)),
+            },
+        };
+
+        let reserved_id = storage_with_eviction_cache
+            .reserve(&Arc::new(RwLock::new(CachedNode::new_clean(node))));
+        assert_eq!(reserved_id, id);
+        assert!(storage_with_eviction_cache.cache.get(&id).is_none());
+    }
+
+    #[test]
+    fn set_inserts_node_into_cache() {
```

Comment:
```suggestion
    fn set_inserts_set_op_into_cache() {
```
You use this naming for the delete op, let's unify them